packages/discord-utilities/src/lib/regexes.ts

```diff
@@ -0,0 +1,118 @@
+ * @remark Capture group 1 can be used to determine whether the emoji is animated or not. It is named `animated`.
+ * @remark Capture group 2 is the name of the emoji as it is typed in a message. It is named `name`.
+ * @remark Capture group 2 is the ID of the emoji. It is named `id`.
+ */
+export const FormattedCustomEmojiWithGroups = /^(?<animated>a?):(?<name>[^:]+):(?<id>\d{17,19})$/;
+
+/**
+ * Regex that matches any URL starting with `http` or `https`
+ * @raw `/^https?:\/\//`
+ * @remark for WebSocket URLs see [[WebsocketGenericUrlRegex]]
+ */
+export const HttpUrlRegex = /^https?:\/\//;
+
+/**
+ * Regex that can capture the Guild, Channel, and Message ID based on a shareable Discord message link.
+ * @raw `/^(?:https:\/\/)?(?:ptb\.|canary\.)?discord(?:app)?\.com\/channels\/(?<guildId>(?:\d{17,19}|@me))\/(?<channelId>\d{17,19})\/(?<messageId>\d{17,19})$/`
+ * @remark Capture group 1 is the ID of the guild the message was sent in. It is named `guildId`.
+ * @remark Capture group 2 is the ID of the channel in that guild the message was sent in. It is named `channelId`.
+ * @remark Capture group 3 is the ID of the message itself. It is named `messageId`.
+ */
+export const MessageLinkRegex = /^(?:https:\/\/)?(?:ptb\.|canary\.)?discord(?:app)?\.com\/channels\/(?<guildId>(?:\d{17,19}|@me))\/(?<channelId>\d{17,19})\/(?<messageId>\d{17,19})$/;
+
+/**
+ * Regex that matches any animated or non-animated custom Discord emoji *without the wrapping `<...>` symbols.
+ * This means that a string that matches this regex can directly be send inside a Discord message.
+ * Other than this difference it is similar to [[FormattedCustomEmoji]].
+ * @raw `/a?:\w{2,32}:\d{17,18}/`
+ */
+export const ParsedCustomEmoji = /a?:\w{2,32}:\d{17,18}/;
+
+/**
+ * Regex that matches any animated or non-animated custom Discord emoji *without the wrapping `<...>` symbols.
+ * This means that a string that matches this regex can directly be send inside a Discord message.
+ * Other than this difference it is similar to [[FormattedCustomEmojiWithGroups]].
+ * @raw `/^(?<animated>a?):(?<name>[^:]+):(?<id>\d{17,19})$/`
+ * @remark Capture group 1 can be used to determine whether the emoji is animated or not. It is named `animated`.
+ * @remark Capture group 2 is the name of the emoji as it is typed in a message. It is named `name`.
+ * @remark Capture group 2 is the ID of the emoji. It is named `id`.
+ */
+export const ParsedCustomEmojiWithGroups = /^(?<animated>a?):(?<name>[^:]+):(?<id>\d{17,19})$/;
+
+/**
+ * Regex that can capture the ID in Discord Role mentions
+ * @raw `/^(?:<@&)?(?<id>\d{17,19})>?$/`
+ * @remark Capture group 1 is the ID of the role. It is named `id`.
+ */
+export const RoleMentionRegex = /^(?:<@&)?(?<id>\d{17,19})>?$/;
+
+/**
+ * Regex that can capture any Discord Snowflake ID
+ * @raw `/^(?<snowflake>\d{17,19})$/`
+ * @remark Capture group 1 is the Snowflake. It is named `snowflake`.
+ */
+export const SnowflakeRegex = /^(?<snowflake>\d{17,19})$/;
```

Comment:
```suggestion
 * @remark Capture group 1 is the Snowflake. It is named `id`.
 */
export const SnowflakeRegex = /^(?<id>\d{17,19})$/;
```

It'd be useful to be able to do:

```javascript
const id = (RoleMentionRegex.exec(str) ?? SnowflakeRegex.exec(str))?.groups.id ?? null;
```

To test a role mention and a ID.